Docker.md

```diff
@@ -0,0 +1,264 @@
+
+## Example Dockerfile for a .NET project
+
+```Dockerfile
+# syntax=docker/dockerfile:1
+# First we create a "build" image
+FROM mcr.microsoft.com/dotnet/sdk:5.0 AS build-env
+# We will be working inside of the /app folder in docker image
+WORKDIR /app
+
+# The general idea behind this step is that you first copy `.csproj` and `.sln` files because they rarely change.
+# Then you run `dotnet restore` which will pull nuget packages.
+# Docker will then understand that if no files changed before `RUN dotnet restore`
+# it can keep using a cached version of the image
+# so you only need to run restore when `.csproj` or `.sln` changes.
+COPY *.sln .
+COPY **/*.csproj ./
+# This is a bash script that will for each `.csproj` create folders named like the project and move
+# the `.csproj` inside of it - it seems strange but the line above this one `COPY **/*.csproj` does not
+# work the way you expect it to - instead of copying the entire folder structure with `.csproj` files
+# it will put all *.csproj files in to the target folder root
+# This is a hacky workaround around an issue in docker, if your project files use nested hierarchy
+# this will not work and you will need to do some other workaround.
+# See this issue for more info : https://github.com/moby/moby/issues/15858
+RUN for file in $(ls *.csproj); do mkdir -p ./${file%.csproj}/ && mv $file ./${file%.csproj}/; done
+RUN dotnet restore
+
+# Now you copy all source files and call build
+COPY . ./
+RUN dotnet publish -c Release -o out
+
+# Now create a runtime image and copy all the build artifacts from build image
+FROM mcr.microsoft.com/dotnet/aspnet:5.0
+WORKDIR /app
+COPY --from=build-env /app/out ./
+
+# Our runtime image will call dotnet Project.Name.API.dll on startup
+ENTRYPOINT ["dotnet", "Project.Name.API.dll"]
+
+```
+
+This docker file will do a 2 stage image build, first it will create a temporary image with .NET SDK called build-env, it will copy the source inside of that image and build it.
+
+Second stage will build the runtime image, this image will only contain the .NET runtime and the built project files.
+
+To create a docker image in your project, add a `Dockerfile` to the root of your solution, then build and run the image :
+
+```bash
+> docker build -t <image-name>
+> docker run --rm -d --name <container-name> <image-name> \
+    -p 8000:80 \
+    -e ASPNETCORE_ENVIRONMENT=Development
+```
+
+You can add additional environment variables for the docker container with more `-e` switches.
+
+## Running ASP.NET core with HTTPS inside of docker
+
+More information on this can be found [here](https://docs.microsoft.com/en-us/aspnet/core/security/docker-compose-https?view=aspnetcore-5.0), but a quick overview
```

Comment:
FROM 
More information on this can be found [here](https://docs.microsoft.com/en-us/aspnet/core/security/docker-compose-https?view=aspnetcore-5.0), but a quick overview
TO
More information on this can be found [here](https://docs.microsoft.com/en-us/aspnet/core/security/docker-compose-https?view=aspnetcore-5.0), but here is a quick overview: